lib/convection/control/cloud.rb

```diff
@@ -16,17 +16,39 @@ def stacks
         @cloudfile.stacks
       end
 
-      def deck
-        @cloudfile.deck
+      def stack_groups
+        @cloudfile.stack_groups
       end
 
-      def converge(to_stack, &block)
-        unless to_stack.nil? || stacks.include?(to_stack)
-          block.call(Model::Event.new(:error, "Stack #{ to_stack } is not defined", :error)) if block
-          return
+      def filter_deck(options = {}, &block)
+        # throw an error if the user specifies both a stack group and list of stacks
+        if options[:stack_group] && options[:stacks]
+          block.call(Model::Event.new(:error, 'Cannot specify --stack-group and --stack-list at the same time', :error)) if block
+          return []
         end
 
-        deck.each do |stack|
+        # throw an error if the user specifies a nonexistent stack groups
+        if options[:stack_group] && !stack_groups.key?(options[:stack_group])
+          block.call(Model::Event.new(:error, "Unknown stack group: #{options[:stack_group]}", :error)) if block
+          return []
+        end
+
+        # throw an error if the user specifies nonexistent stacks
+        if Array(options[:stacks]).any? { |name| !@cloudfile.stacks.key?(name) }
+          bad_stack_names = options[:stacks].reject { |name| @cloudfile.stacks.key?(name) }
+          block.call(Model::Event.new(:error, "Non Existent Stack(s) #{bad_stack_names.join(', ')}", :error)) if block
```

Comment:
Should we be using "Undefined" instead of "Non Existent"? If a stack is defined in your Cloudfile but has not been converged yet then the create call to CloudFormation wouldn't have been called meaning there is no matching stack in cloud formation.